### DISCUSSION TITLE
TurbSim mean wind speed from Uref

### DISCUSSION THREAD

**Discussion by lt72884** [OpenFAST/openfast#1723] [Q&A]
I have a question. When i generate a turbsim.bts file and i have Uref set to 10 m/s, it generates a file which has an average windspeed of 10.2087 m/s.

Not changing anything in the TurbSim.inp file, and leaving the Uref to 10m/s, and then generating a new .bts file, i still get 10.2087 as the average. 

Is there anyway to keep the Uref at 10m/s but have it generate a different average? IE, 10.3654 or 10.1327 etc? 

Does changing the random number do anything to windspeed at all?

thanks

**Comment by jjonkman** [comment]
Dear @lt72884,

TurbSim superimposes the effect of the mean wind profile (shear, veer as function of height) and turbulent fluctuations (with zero mean).  The random seeds only effect the turbulence fluctuations, not the means.  I'm not sure I fully understand your question (it would be helpful to see your input file and what you change between simulations), but the hub-height mean wind speed can certainly be different than the reference wind speed if the reference height is not the height of the hub.

Also, if you are not outputting the full time series generated internally within TurbSim (`UsableTime` < `AnalysisTime`), it could also be that the turbulence fluctuations do not average out to zero over the output time window.

Best regards,

**Comment by lt72884** [comment]
I have not changed anything in the input file between simulations using TurbSim. 

What my question is, how do i get it to generate a different mean wind speed? It always produces the exact same speed no matter how many times i run the simulation of TurbSim using 10m/s as my Uref.

it always generates 10.2087 m/s. 

Well, my useable time is 60 seconds and analysis time is 630 seconds. I did a small test of changing the random seed to another 6 digit number and it created a differnt average speed which is what im looking for, but may not be the best approach.

What im trying to do is have a bunch of .bts files that all have 10m/s as the current windpseed but not have the average is not the same value of 10.2087 because in reality, wind changes all the time. I am wanting to create 10 different bts files with 10m/s wind speed but get back 10 different averages if that makes sense haha.

thanks

**Comment by bjonkman** [comment]
If you request an average wind speed of 10 m/s, you will get an average of 10 m/s (when you average across the full analysis time that is generated). The different random seeds will produce different **instantaneous** values of wind speed, which is why the average of the first 60 seconds of data changes. However, the average across the full 630 seconds of analysis time should not change.

**Comment by lt72884** [comment]
interesting. thanks for that information. 

i have changed the useable time to 300 seconds and my average windspeed is 9.6785 mps. If i tack on an extra number on the random seed, the new value is 9.8743 . this matches with what you say in your reply to me.

i will set the usable time to be the full 630 and change the seed number and see what happens. i am curious. 

thanks
